### PR TITLE
Update build-manifest

### PR DESCRIPTION
**Pull request by cjyclaire**
This updates service build with directory name as key instead of endpointPrefix.

cc: @jeskew @xibz @mtdowling

### CODE REVIEW

File: build/build-manifest.php [hunk 1]
```diff
@@ -51,8 +51,10 @@
 $manifest = [];
 foreach (glob(__DIR__ . '/../src/data/**/**/api-2.json') as $file) {
     $model = json_decode(file_get_contents($file), true);
+    preg_match('@src/data/([^/]+)/[0-9]{4}-[0-9]{2}-[0-9]{2}/api-2.json$@', $file, $matches);
+    $identifier = $matches[1];
```
Comment (xibz) [inline build/build-manifest.php]: Should we handle when the regex doesn't match?

Comment (cjyclaire) [inline build/build-manifest.php]: That's true, I'm also thinking about testing it throughly, will follow up with a new commit.

**Comment by cjyclaire** [comment]
Inject `serviceIdentifier` to metadata to make sure that Paginators, Waiters and docs loaded right.

**Comment by xibz** [comment]
LGTM, :shipit:

**Pull request merged**